test/compflags/link/sungeun/static_dynamic.skipif

```diff
@@ -6,6 +6,15 @@ if [[ $CHPL_HOST_PLATFORM == *cygwin* \
   # Can't do static/dynamic linking on these platforms
   #
   echo True
+elif [[ $CHPL_COMM_SUBSTRATE == mpi \
+        || ( $CHPL_COMM == ofi && $CHPL_LAUNCHER == mpirun4ofi ) ]] ; then
```

Comment:
Seems like the `CHPL_COMM == ofi` check shouldn't be needed, but I don't mind keeping it.